source/lib.js

```diff
@@ -140,15 +140,199 @@ function gatherRecursiveWithinFunction(node, predicate) {
   return gatherRecursive(node, predicate, isFunction)
 }
 
+// Trims the first single space from the spacing array or node's children if present
+// maintains $loc for source maps
```

Comment:
This also inserts string `c` after removing the first space or not but we only ever use it with `""`. We may want to rename it later. I think it was originally used to insert an open paren while removing a space if present but we do that a different way now.

Reply:
I did actually find one instance of `insertTrimmingSpace("case ", ws)`. But I agree it's a pretty confusing name given that it's mostly trimming, not inserting. Should we just call it `trimSpace` or `trimLeadingSpace`, and make `""` the default value for the second argument?

(I'm going to merge this PR given that this was just a function move, but can address this comment in a future PR.)